workflow/controller/workflowpod.go

```diff
@@ -696,7 +696,7 @@ func isResourcesSpecified(ctr *apiv1.Container) bool {
 
 // addMetadata applies metadata specified in the template
 func (woc *wfOperationCtx) addMetadata(pod *apiv1.Pod, tmpl *wfv1.Template) {
-	if woc.wf.Spec.PodMetadata != nil {
+	if woc.execWf.Spec.PodMetadata != nil {
 		// add workflow-level pod annotations and labels
 		for k, v := range woc.wf.Spec.PodMetadata.Annotations {
```

Comment:
here the another place `woc.wf.Spec.PodMetadata.Annotations` to `woc.execwf.Spec.PodMetadata.Annotations`